mongo/description/version_range.go

```diff
@@ -25,6 +25,17 @@ func (vr VersionRange) Includes(v int32) bool {
 	return v >= vr.Min && v <= vr.Max
 }
 
+// Equals returns a bool indicating whether the supplied VersionRange is equal.
+func (vr *VersionRange) Equals(other *VersionRange) bool {
+	if vr == nil && other == nil {
+		return true
+	}
+	if vr == nil || other == nil {
+		return false
+	}
+	return vr.Min == other.Min && vr.Max <= other.Max
```

Comment:
Why is the check `vr.Max <= other.Max`? If the server's `maxWireVersion` changes, it means the server has either upgraded or downgraded. That definitely seems like a situation where the new description would not be equal to the old one.